devito/logger.py

```diff
@@ -86,6 +86,27 @@ def set_log_level(level, comm=None):
     configuration['log-level'] = level
 
 
+class switch_log_level(object):
+    """
+    A context manager to temporarily change MPI logging.
+    """
+
+    def __init__(self, comm):
+
+        from devito import configuration
+        self.level = configuration['log-level']
+        self.comm = comm
+
+    def __enter__(self):
+        # Limit logging to rank 0
+        set_log_level(self.level, self.comm)
+        return
+
+    def __exit__(self, *args):
+        set_log_level(self.level)
+        logger.addHandler(stream_handler)
```

Comment:
shouldn't this line go inside `set_log_level` in an `else` branch representing the `comm is None` case? this way, a `set_log_level` in user code will behave as expected

Reply:
note: the above is nitpicking